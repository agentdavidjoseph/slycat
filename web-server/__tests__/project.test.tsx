
import React from 'react';
import Project, 
{ProjectProps, ProjectState} 
from "components/Project";
import { mount } from "enzyme";


describe('when loading a bad marking the UnrecognizedMarkingWarning',() =>{ 
  const props = {
    name: "test",
    id: "1234",
    description: "test",
    created: "1234",
    creator: "me"
  }
  const render = mount(
    <Project {...props} />
  );
  test('we have no props on initial load', () => {
    const props = render.props() as ProjectProps;
    expect(props).toMatchSnapshot();
  });

  test('we no state', () => {
    const state = render.state() as ProjectState;
    expect(state).toMatchSnapshot();
  });

  test('we should have a full initialization', () => {
    expect(render).toMatchSnapshot();
  });

});

export default undefined