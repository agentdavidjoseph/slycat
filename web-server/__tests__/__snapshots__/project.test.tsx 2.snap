// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`when loading a bad marking the UnrecognizedMarkingWarning we have no props on initial load 1`] = `
Object {
  "created": "1234",
  "creator": "me",
  "description": "test",
  "id": "1234",
  "name": "test",
}
`;

exports[`when loading a bad marking the UnrecognizedMarkingWarning we no state 1`] = `null`;

exports[`when loading a bad marking the UnrecognizedMarkingWarning we should have a full initialization 1`] = `
<Project
  created="1234"
  creator="me"
  description="test"
  id="1234"
  name="test"
>
  <a
    className="list-group-item list-group-item-action"
    href="/projects/1234"
  >
    <div
      className="h6"
    >
      <span
        className="badge badge-secondary mr-1"
      >
        project
      </span>
       
      test
    </div>
    <p
      className="mb-2"
    >
      test
    </p>
    <small>
      <em>
        Created 
        <span>
          1234
        </span>
         by 
        <span>
          me
        </span>
      </em>
    </small>
  </a>
</Project>
`;
